kernel/src/scan/mod.rs

```diff
@@ -320,10 +321,36 @@ pub(crate) enum TransformExpr {
     Partition(usize),
 }
 
-// TODO(nick): Make this a struct in a follow-on PR
-// (data, deletion_vec, transforms)
-pub type ScanData = (Box<dyn EngineData>, Vec<bool>, Vec<Option<ExpressionRef>>);
+/// Result of a data scan operation containing filtered data and associated transformations.
+pub struct ScanData {
+    /// Engine data with its selection vector indicating relevant rows
+    pub filtered_data: FilteredEngineData,
```

Comment:
comment improvement and rename (we know it's filtered engine data from the type, so we can give it a semantic name with the binding)
```suggestion
    /// Filtered engine data with one row per file to scan (and only selected rows should be scanned)
    pub scan_files: FilteredEngineData,
```